app/models/course/condition.rb

```diff
@@ -5,4 +5,6 @@ class Course::Condition < ActiveRecord::Base
   belongs_to :conditional, polymorphic: true
 
   delegate :satisfied_by?, to: :actable
+
+  NAMES = ['achievement', 'assessment', 'level']
```

Comment:
Then write `ALL_CONDITIONS = [Course::Achievement::Condition.name, ...]` or something to that effect.

If you're feeling adventurous, submit a PR to acts_as